### PR TITLE
Add ClientResourceServerAssociation class

### PR DESCRIPTION
**Pull request by jerriep**
Add ClientResourceServerAssociation class for associating resource servers with clients

This is fix for #147

### CODE REVIEW

**Comment by jerriep** [comment]
Seem we should use https://auth0.com/docs/api/management/v2#!/Client_Grants/post_client_grants instead. Will update

File: src/Auth0.ManagementApi/Models/ClientBase.cs [hunk 1]
```diff
@@ -119,9 +119,11 @@ public abstract class ClientBase
         [JsonProperty("name")]
         public string Name { get; set; }
 
-
+        /// <summary>
+        ///
```
Comment (nicosabena) [inline src/Auth0.ManagementApi/Models/ClientBase.cs]: How about `A list of resource servers (APIs) that the client is authorized to request access tokens for, using the Client Credentials exchange.`?

File: src/Auth0.ManagementApi/Models/ClientResourceServerAssociation.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    public class ClientResourceServerAssociation
+    {
+        /// <summary>
+        /// The resource server identifier.
+        /// </summary>
+        [JsonProperty("identifier")]
+        public string Identifier { get; set; }
+
+        /// <summary>
+        /// The scopes to request from the user to access the resource server.
```
Comment (nicosabena) [inline src/Auth0.ManagementApi/Models/ClientResourceServerAssociation.cs]: I'd say it's more like "The scopes granted to the client to access the resource server."

File: src/Auth0.ManagementApi/Models/ClientResourceServerAssociation.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Auth0.ManagementApi.Models
+{
+    public class ClientResourceServerAssociation
+    {
+        /// <summary>
+        /// The resource server identifier.
```
Comment (nicosabena) [inline src/Auth0.ManagementApi/Models/ClientResourceServerAssociation.cs]: I would add ` (API)` after `The resource server`, to match the dashboard and docs more closely.

**Pull request merged**